lib/feed-v1/validation.js

```diff
@@ -36,37 +36,56 @@ function validateShape(msg) {
 
 function validateWho(msg) {
   try {
-    base58.decode(msg.metadata.who)
+    const whoBuf = base58.decode(msg.metadata.who)
+    if (whoBuf.length !== 32) {
+      return new Error(
+        `invalid message: decoded "who" should be 32 bytes but was ${whoBuf.length}`
+      )
+    }
   } catch (err) {
     return new Error('invalid message: must have "who" as base58 string')
   }
 }
 
 function validateMsgHash(str) {
   try {
-    base58.decode(str)
+    const hashBuf = Buffer.from(base58.decode(str))
+    if (hashBuf.length !== 16) {
+      return new Error(
+        `invalid message: decoded hash should be 16 bytes but was ${hashBuf.length}`
+      )
+    }
   } catch (err) {
     return new Error(
       `invalid message: msgHash ${str} should have been a base58 string`
     )
   }
 }
 
+function validateSize(msg) {
+  const {
+    metadata: { size },
+  } = msg
+  if (!Number.isSafeInteger(size) || size < 0) {
+    return new Error(`invalid message: "size" should be an unsigned integer`)
+  }
+}
```

Comment:
Ideally this should also measure the size of the `content` and match with `metadata.size`.

But it can be a TODO for later :)